### PR TITLE
feat: export domhandler's Element to support instanceof checks

### PR DESCRIPTION
**Pull request by natterstefan**
## What is the motivation for this pull request?

It's a feature that resolves what we have [discussed here](https://github.com/remarkablemark/html-react-parser/issues/252#issuecomment-896406971) (and [suggested by others as well](https://github.com/remarkablemark/html-react-parser/issues/199#issuecomment-881463160)).

## What is the current behavior?

Currently, one has to [import `Element` by from `domhandler`](https://github.com/remarkablemark/html-react-parser/issues/221#issuecomment-771600574) manually, after adding it to the project, while it is already a `dependency` of this package.

## What is the new behavior?

`html-react-parser` exports the `Element` type of `domhandler` already, but not the runtime version of it. This PR fixes it.

## Checklist:

- [x] Tests
- [x] Documentation
- [x] Types

I also added another CRA example for TypeScript and linked it in the `README.md` file.

## Related Issues

resolves https://github.com/remarkablemark/html-react-parser/issues/252
resolves https://github.com/remarkablemark/html-react-parser/issues/221 (most likely too)
resolves https://github.com/remarkablemark/html-react-parser/issues/199

### CODE REVIEW

File: examples/create-react-app-typescript/yarn.lock [hunk 1]
```diff
@@ -0,0 +1,11696 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
```
Comment (remarkablemark) [inline examples/create-react-app-typescript/yarn.lock:2]: Can we keep either `yarn.lock` or `package-lock.json` for CRA example? Feels redundant to have both

File: index.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 var domToReact = require('./lib/dom-to-react');
 var attributesToProps = require('./lib/attributes-to-props');
 var htmlToDOM = require('html-dom-parser');
+var { Element } = require('domhandler/lib/node');
```
Comment (remarkablemark) [inline index.js:4]: Since this file isn't compiled, can you update so it doesn't destructure?

File: index.js [hunk 3]
```diff
@@ -31,6 +32,7 @@ function HTMLReactParser(html, options) {
 HTMLReactParser.domToReact = domToReact;
 HTMLReactParser.htmlToDOM = htmlToDOM;
 HTMLReactParser.attributesToProps = attributesToProps;
+HTMLReactParser.Element = Element;
```
Comment (remarkablemark) [inline index.js:35]: Following up on my previous comment, I'm ok if you did this:

```js
HTMLReactParser.Element = require('domhandler/lib/node').Element;
```

**Review by remarkablemark — CHANGES_REQUESTED** [review]
Thanks for opening this PR @natterstefan, I left some comments but otherwise it looks good!

File: examples/create-react-app-typescript/yarn.lock (hunk 1 above)
Comment (natterstefan) [reply examples/create-react-app-typescript/yarn.lock:2]: Oh, I am sorry. I didn't realize that there are too. I'll take care of it. I suggest adding once of those in the main folder anyway as well.

File: index.js (hunk 2 above)
Comment (natterstefan) [reply index.js:4]: Sure, thanks for the feedback. I can do that for sure.

Comment (natterstefan) [reply index.js]: See https://github.com/remarkablemark/html-react-parser/pull/296/commits/f57d7da7e15e0e71ed2bf69a4a0219d8c5b5adfc, please.

**Review by remarkablemark — APPROVED** [review]
LGTM

**Comment by remarkablemark** [comment]
[Size Limit](https://github.com/remarkablemark/html-react-parser/pull/296/checks?check_run_id=3763426643#step:3:102):

```sh
index.js → dist/html-react-parser.min.js...
created dist/html-react-parser.min.js in 975ms
/usr/local/bin/npx size-limit --json
[
  {
    "name": "dist/html-react-parser.min.js",
    "passed": true,
    "size": 9226,
    "running": 0.07007999999999999,
    "loading": 0.1801953125
  }
]
```

**Comment by remarkablemark** [comment]
Release #306

**Pull request merged**